src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
-                "doc": "Augmented table containing extracted metadata."
+                "doc": "Augmented table containing extracted metadata.",
             }
         }
 
     def process(self, inputs, outputs) -> None:
-        
+        import re
+        import polars as pl
+        import pyarrow as pa
+
         table_obj = inputs.get_value_obj("corpus_table")
         column_name = inputs.get_value_obj("file_name_col").data
-        pub_refs, pub_names = None, None
 
-        try:
-            pub_refs = inputs.get_value_obj("map").data[0]
-            pub_names = inputs.get_value_obj("map").data[1]
+        if not column_name:
+            raise KiaraProcessingException("No file name column name provided.")
+        
+        if not table_obj:
+            raise KiaraProcessingException("No table provided.")
+        
+        if table_obj.is_set:
 
-        except Exception as e:
-            pass;
+            sources: Union[pa.Table, None] = table_obj.data
+            
+            assert sources is not None
 
+            sources_col_names = sources.column_names
+
+            if column_name not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find file names column '{column_name}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            sources_data: pa.Table = table_obj.data.arrow_table
+            
+            sources_tb: pl.DataFrame = pl.from_arrow(sources_data) # type: ignore
 
-        sources = pl.from_arrow(table_obj.data.arrow_table)
 
         def get_ref(file):
             try:
-                ref_match = re.findall(r'(\w+\d+)_\d{4}-\d{2}-\d{2}_',file)
+                ref_match = re.findall(r"(\w+\d+)_\d{4}-\d{2}-\d{2}_", file)
                 if not ref_match:
                     return None
                 return ref_match[0]
-            
+
             except Exception as e:
-                print(f"Error in get_ref: {e}")
-                return None
+                raise KiaraProcessingException(e)
 
         def get_date(file):
             try:
-                date_match = re.findall(r'_(\d{4}-\d{2}-\d{2})_',file)
+                date_match = re.findall(r"_(\d{4}-\d{2}-\d{2})_", file)
                 if not date_match:
                     return None
                 return date_match[0]
             except Exception as e:
-                print(f"Error in get_date: {e}")
-                return None
+                msg = f"Error in get_date: {e}"
+                raise KiaraProcessingException(msg)
 
+        
         try:
-            sources = sources.with_columns([
-                sources[column_name].apply(get_date).alias('date'),
-                sources[column_name].apply(get_ref).alias('publication_ref')
-            ])
-        except Exception as e:
+            
+            pub_refs: list[str] = inputs.get_value_obj("map").data[0]
```

Comment:
Ok, here is a good example where you needed `is_set` Because you marked 'map' as optional, the `.data` property could be None, and then you would get a runtime error because `None` can't be indexed with `[0]`. I guess you could just remove the 'optional' from the `map` input though, right? I see you catch an exception, so maybe it's not necessary, but for someone like me it's not obvious what exactly you expect to go wrong in this piece of code, so if you are happy with this just failing then you should probably add a few comments.

But I think the better way to do it would be to test whether 'map' is set, and use a conditional path to handle that explicitely. In this case, you'd also have to test that the 'map' is not empty, because if it was, the `[0]` indexed element access would raise another error. Same goes for the line below, but won't add another comment for that since it'd be the same.